content/posts/a-guide-on-commit-messages.mdx

```diff
@@ -0,0 +1,169 @@
+
+We can see sometime a capitalized message. There are metadata for only two commits. Finally some messages content are so light we can't figure out what the commit does. In the same article Chris goes through 7 rules to write a good commit message.
+
+> 1. Separate subject from body with a blank line
+> 2. Limit the subject line to 50 characters
+> 3. Capitalize the subject line
+> 4. Do not end the subject line with a period
+> 5. Use the imperative mood in the subject line
+> 6. Wrap the body at 72 characters
+> 7. Use the body to explain what and why vs. how
+>
+> Chris Beams - [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/)
+
+I personnally apply a subset of these rules:
+
+- Limit the subject line to 50 characters
+- Capitalize the subject line
+- Use the imperative mood in the subject line
+
+The character limit does not only force you to be concise which is a great skill but also fits well with a lot of tools as explained in Tim Pope article [A Note About Git Commit Messages](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html).
+
+I tend to think the perfect is the enemy of the good. That is why I omitted talking about commit messages body. I think we have enough work to do on the summary line. But I would add one rule as counterpart:
+
+- Add a pull request/merge request tracking ID
+
+Most of the time, we use tools to discuss during a code review process like GitHub, Gitlab, ... All pieces of information about the change will be at least described there. I added this rule about tracking ID so it becomes easy to find related pull/merge request. Writing a commit body is better than adding tracking IDs because if you change your repository manager solution you will loose all this precious context information and your tracking IDs will be useless.
+
+Here is a more recent example from a project I work on called [gitmoji-changelog](https://github.com/frinyvonnick/gitmoji-changelog) that follow these rules:
+
+```
+Make core independent from the git client (#171)
+Upgrade Docker image version (#167)
+Add maven preset (#172)
+Add a generic preset using configuration file (#160)
+Improve error messages for preset system (#161)
+Publish Canary version on master push (#141)
+```
+
+Having a consistent style makes it much more easier to read. You can figure out the context of the change and metadata are directly accessible. The most important thing is to find common rules in your team to ensure having a well structured commit history. There are some conventions that can help you finding these but we will come back to this subject later.
+
+If you have difficulties to find out what message you should write maybe you should split your commit in smaller parts. Peter Hutterer listed in his article [On commit messages](http://who-t.blogspot.com/2009/12/on-commit-messages.html) a few examples where it is difficult to find out a good commit message because your commit patch may not be logical.
```

Comment:
```suggestion
If you have difficulties to find out what message you should write, maybe you should split your commit in smaller parts. Peter Hutterer listed in his article [On commit messages](http://who-t.blogspot.com/2009/12/on-commit-messages.html) a few examples where it is difficult to find out a good commit message because your commit patch may not be logical.
```